File: src/main/java/com/example/domain/member/dto/response/MyReviewItemResponse.java
package com.example.domain.member.dto.response;

import com.example.domain.review.entity.Review;

import java.time.LocalDateTime;

public record MyReviewItemResponse(
        Long reviewId,
        Long festivalId,
        String festivalTitle,
        Integer rating,
        String content,
        String reviewImageUrl,
        Integer likeCount,
        LocalDateTime createdAt
) {
    public static MyReviewItemResponse from(Review review){
        return new MyReviewItemResponse(
                review.getId(),
                review.getFestival().getId(),
                review.getFestival().getTitle(),
                review.getRating(),
                review.getContent(),
                review.getImage(),
                review.getLikeCount(),
                review.getCreatedAt()
        );
    }
}

File: src/main/kotlin/com/example/domain/member/dto/response/MyReviewItemResponse.kt
package com.example.domain.member.dto.response

import com.example.domain.review.entity.Review
import java.time.LocalDateTime


data class MyReviewItemResponse(
    val reviewId: Long,
    val festivalId: Long,
    val festivalTitle: String,
    val rating: Int,

    val content: String,
    val reviewImageUrl: String?,
    val likeCount: Int,
    val createdAt: LocalDateTime
) {
    companion object {
        fun from(review: Review): MyReviewItemResponse {
            return MyReviewItemResponse(
                review.id,
                review.festival.id,
                review.festival.title,
                review.rating,
                review.content,
                review.image,
                review.likeCount,
                review.createdAt
            )
        }
    }
}
